### ISSUE TITLE
how to handle `after_initialize`?

### ISSUE THREAD

**Issue by breathe** [zdennis/activerecord-import#690]
I have a rails model with an `after_initialize` method that enforces some non-nil fields as early as possible in the lifecycle of the program ...

```
class GqlLocationInfo < ActiveRecord::Base

  # ugh -- field named 'type' is special in activerecord ...
  # disable that behavior so that field behaves as regular column
  self.inheritance_column = nil

  after_initialize :new_record_defaults, :if => :new_record?

  private

  def new_record_defaults
    if self.type.nil?
      raise ArgumentError, "Invariant violation: type must be provided"
    end

    if self.revision.nil?
      raise ArgumentError, "Invariant violation: revision must be provided"
    end

    if self.location_id.nil?
      raise ArgumentError, "Invariant violation: location_id must be provided"
    end

    self.zone = GqlLocation::zone

    if self.zone.nil?
      raise ArgumentError, "Invariant violation: zone cannot be nil"
    end

    self.version = GqlLocation::current_storage_version(for_type: self.type)

    if self.version.nil?
      raise ArgumentError, "Invariant violation: version cannot be nil"
    end
  end

```

Is there anyway to use such a model with activerecord-import?

I've tried with ar records and hashes and I always get failures inside of new_record_defaults -- there is some point within active-record-import where the model above is instantianted with no properties set at all ...

**Comment by AhMohsen46** [comment]
after_initialize was being called on initializing objects in [import.rb#L998](https://github.com/zdennis/activerecord-import/blob/c7660a488ea76fb55182b3ce05b514fd357c866c/lib/activerecord-import/import.rb#L998) and [import.rb#L579](https://github.com/zdennis/activerecord-import/blob/c7660a488ea76fb55182b3ce05b514fd357c866c/lib/activerecord-import/import.rb#L579)
as calling Model.new was initializing a new object with no attributes, and since you were validating on attributes presence, an error was being raised
to solve this I've created this [PR#691](https://github.com/zdennis/activerecord-import/pull/691)
where I've used [allocate](https://apidock.com/ruby/Class/allocate) instead of [new](https://apidock.com/ruby/Class/new)
the difference is that `new` was calls `allocate` then calls initialize, since `initialize` doesn't serve the purpose of fetching the timestamps_attributes and runs the `after_initialize` callback, which causes the error, the PR uses allocate instead for this purpose

**Comment by jkowens** [comment]
@AhMohsen46 nice work!

**Comment by AhMohsen46** [comment]
Thanks :D

**Comment by breathe** [comment]
Cool!  

Sorry to bug -- @jkowens -- will this be a new release of gem with this PR?  Would like to have this functionality :)

**Comment by jkowens** [comment]
@breathe sorry for the delay, I'll try to cut a new release this weekend.

**Comment by jkowens** [comment]
This has been released in v1.0.6 🎉
